src/main/scala/is/hail/rvd/UnpartitionedRVD.scala

```diff
@@ -52,7 +52,7 @@ class UnpartitionedRVD(val rowType: TStruct, val crdd: ContextRDD[RVDContext, Re
     UnpartitionedRVDSpec(rowType, codecSpec, partFiles)
 
   def coalesce(maxPartitions: Int, shuffle: Boolean): UnpartitionedRVD =
-    new UnpartitionedRVD(rowType, crdd.coalesce(maxPartitions, shuffle = shuffle))
+    new UnpartitionedRVD(rowType, stably(_.coalesce(maxPartitions, shuffle = shuffle)))
```

Comment:
Somethings a bit wonky here.

Coalesce has two cases: First, shuffle.  In that case, you should use the shuffle logic which serializes and stably isn't necessary.  Second, you scan and compute which partitions to coalesce, and then there is a no need to leave ContextRDD land, you're just merging partitions.

Reply:
Also, right now OrderedRVD is doing two back-to-back shuffles, which seems ... excessive.

Reply:
Where do you see the back-to-back shuffles? I thought I saw that at one point but I've forgotten where it was.

Reply:
Also, yes, I'll figure out how to handle these two cases.